ui/src/app/workflows/components/workflow-details/artifact-panel.tsx

```diff
@@ -43,12 +45,25 @@ export const ArtifactPanel = ({
     useEffect(() => {
         setObject(null);
         setError(null);
+        setShowDownloadLink(true);
         if (ext === 'json') {
             requests
                 .get(services.workflows.artifactPath(workflow, artifact.nodeId, artifact.name, archived, input))
                 .then(r => r.text)
                 .then(setObject)
                 .catch(setError);
+        } else {
```

Comment:
I don't think you need this. The iframe will show "Not found". I think that is enough.

Reply:
In fact, I don't think you need any UI changes.

Reply:
Oh, I thought you wanted a friendly message here? The story says to show "Artifact has been deleted". But yes, I can see how showing the default "File not found" message would be sufficient.

Reply:
I think that should be fine, because they will get an error. What you don't want to do is download a large file twice.

Reply:
I guess it won't cache it, will it?

Reply:
Another possibility is to set the iFrame content equal to the result of the earlier request. But perhaps that can be for another day.